azkaban-web-server/src/main/java/azkaban/scheduler/QuartzJobDescription.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2017 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package azkaban.scheduler;
+
+import java.io.Serializable;
+import java.util.Map;
+
+/**
+ * Manage one quartz job's variables. Every AZ Quartz Job should come with a QuartzJobDescription.
+ */
+public class QuartzJobDescription<T extends AbstractQuartzJob> {
+
+  private final String groupName;
+  private final Class<T> jobClass;
+  private final Map<String, ? extends Serializable> contextMap;
+
+  public QuartzJobDescription(final Class<T> jobClass,
+      final String groupName,
+      final Map<String, ? extends Serializable> contextMap) {
+
+    /**
+     * This check is necessary for raw type. Please see test
+     * {@link QuartzJobDescriptionTest#testCreateQuartzJobDescription2}
+     */
+    if (jobClass.getSuperclass() != AbstractQuartzJob.class) {
+      throw new RuntimeException("jobClass must extend AbstractQuartzJob class");
```

Comment:
i would prefer a more meaningful exception type, maybe like illegalargumentexception?

Reply:
I found that CastException seems more appropriate. Updated